src/components/items.tsx

```diff
@@ -5,43 +5,81 @@ import { v4 as uuidv4 } from "uuid";
 
 export function Items() {
   const [input, setInput] = useState("");
-  const [things, setThings] = useState<TItemDocument[]>([]);
-  useEffect(() => {
+  const [things, setThings] = useState<Partial<TItemDocument>[]>([]);
+
+  const fetchItems = () =>
     itemsServices.listItems().then((res) => {
       setThings(res);
     });
+  useEffect(() => {
+    fetchItems();
   }, []);
   const addItem = () => {
     const $id = uuidv4();
     const newItem = {
       name: input,
       isSelected: false,
     };
+    setThings((prev) => [...prev, { $id, ...newItem }]);
+
     itemsServices.newItem($id, newItem);
     setInput("");
   };
+  const updateItem = (t: Partial<TItemDocument>) => {
+    t.isSelected = !t.isSelected!;
+    const newArray = things.filter((i) => i.$id !== t.$id);
+    const index = things.findIndex((i) => i.$id === t.$id);
+    newArray.splice(index, 0, t);
+    setThings([...newArray]);
+    if (t.$id && t.name)
+      itemsServices.updateItem(t.$id, {
+        name: t.name,
+        isSelected: t.isSelected,
+      });
+  };
   const deleteItem = (id: string) => {
     itemsServices.deleteItem(id);
+    const newArray = things.filter((i) => i.$id !== id);
+    setThings(newArray);
   };
-  console.log(things);
   return (
     <div className="itemsPage">
-      <div className="inputsField">
-        <input
-          type="text"
-          value={input}
-          onChange={(e) => setInput(e.target.value)}
-        />
-        <button onClick={addItem}>Add</button>
+      <div>
+        <div className="title">
+          <h1>Maria's List</h1>
+        </div>
+        <form
+          className="inputsField"
+          onSubmit={(e) => {
+            e.preventDefault();
+            addItem();
+          }}
+        >
+          <input
+            type="text"
+            value={input}
+            onChange={(e) => setInput(e.target.value)}
+          />
+          <button type="submit">Add</button>
+        </form>
       </div>
       <ul>
         {things.map((t) => (
-          <div className="newItem">
-            <input type="checkbox" id="isSelected" />
+          <div className={`newItem ${t.isSelected ? "checked" : "notChecked"}`}>
```

Comment:
Each element should have an unique `key` prop
```suggestion
          <div key={t.$id} className={`newItem ${t.isSelected ? "checked" : "notChecked"}`}>
```